### ISSUE TITLE
Problems about iou loss

### ISSUE THREAD

**Issue by zhoulukuan** [open-mmlab/mmdetection#2518]
In this `Generalized Intersection over Union: A Metric and A Loss for Bounding Box
Regression` and `Generalized Intersection over Union: A Metric and A Loss for Bounding Box Regression`(giou), I found: Liou = 1 - iou.
However, in mmdetection, I found such codes:
```
@weighted_loss
def iou_loss(pred, target, eps=1e-6):
    """IoU loss.

    Computing the IoU loss between a set of predicted bboxes and target bboxes.
    The loss is calculated as negative log of IoU.

    Args:
        pred (Tensor): Predicted bboxes of format (x1, y1, x2, y2),
            shape (n, 4).
        target (Tensor): Corresponding gt bboxes, shape (n, 4).
        eps (float): Eps to avoid log(0).

    Return:
        Tensor: Loss tensor.
    """
    ious = bbox_overlaps(pred, target, is_aligned=True).clamp(min=eps)
    loss = -ious.log()
    return loss
```
I want to know how to compute iou loss and which one is better?

**Comment by hellock** [comment]
There have been two forms of IoULoss in different papers. We have not compared which one is better under the same settings. (most likely there will not be much difference)

**Comment by zhoulukuan** [comment]
@hellock Thanks a lot. Can you tell me which paper uses log iou loss?

**Comment by hellock** [comment]
UnitBox: An Advanced Object Detection Network

**Comment by hyz-xmaster** [comment]
Hi @hellock，following this thread, I am also having a question about the IoU loss used in the FCOS training. I noticed that [IoU loss](https://github.com/open-mmlab/mmdetection/blob/cd4e716b0470bde842c20c00f18cffa20c2673c1/mmdet/models/anchor_heads/fcos_head.py#L193-L198) there is weighted by the centerness targets. 

So what's consideration of this weighting? Thank you.
